packages/apidom-converter/src/strategies/openapi-3-1-to-openapi-3-0-3/refractor-plugins/security-scheme-type.ts

```diff
@@ -0,0 +1,65 @@
+import {
+  ComponentsElement,
+  OpenApi3_1Element,
+  SecurityRequirementElement,
+  SecuritySchemeElement,
+} from '@swagger-api/apidom-ns-openapi-3-1';
+import { AnnotationElement, toValue, cloneShallow } from '@swagger-api/apidom-core';
+
+type SecuritySchemeTypePluginOptions = {
+  annotations: AnnotationElement[];
+};
+
+const securitySchemeTypeRefractorPlugin =
+  ({ annotations }: SecuritySchemeTypePluginOptions) =>
+  () => {
+    const removedSecuritySchemes: SecuritySchemeElement[] = [];
+    const annotation = new AnnotationElement(
+      '"mutualTLS" type is not supported in OpenAPI 3.0.3. It will be removed from the converted document.',
+      { classes: ['error'] },
+      { code: 'mutualTLS' },
+    );
+
+    return {
+      visitor: {
+        OpenApi3_1Element(element: OpenApi3_1Element) {
```

Comment:
Here is the version of `OpenApi3_1Element` I'm proposing:

```ts
import { isArrayElement } from '@swagger-api/apidom-core';
import { isComponentsElement, isSecuritySchemeElement } from '@swagger-api/apidom-ns-openapi-3-0/src';

OpenApi3_1Element(element: OpenApi3_1Element) {
  if (!isComponentsElement(element.components)) return undefined;
  if (!isArrayElement(element.components.securitySchemes)) return undefined;

  element.components.securitySchemes.forEach((item) => {
    if (isSecuritySchemeElement(item) && toValue(securityScheme.type) === 'mutualTLS') {
      removedSecuritySchemes.push(securityScheme);
    }
  });
}
```

Reasoning:

1. we cannot assume that `element?.components?.securitySchemes` is array. It can be a string or null. We parse the document as it is, we don't apply any validation. So we need to do assertion on everything.
2. again we cannot assume that `const securityScheme = el as SecuritySchemeElement;` is Security Scheme. It can be null, undefined, or anything else.

---

> NOTE: `isSecuritySchemeElement` was added in https://github.com/swagger-api/apidom/pull/3810

Reply:
Thanks for feedback. I added those missing checks. However it seems that `isObjectElement` should be used for `element.components.securitySchemes` as `securitySchemes` has type `ObjectElement | undefined` and `isArrayElement` for `securitySchemes` never returns `true`.

Reply:
> has type ObjectElement | undefined

Yes, you're absolutely right, I did a mistake there. Of course it's an object.